src/api/uploads/MultiputUpload.js

```diff
@@ -329,6 +381,171 @@ class MultiputUpload extends BaseMultiput {
         this.processNextParts();
     }
 
+    /**
+     * Resume uploading the given file
+     *
+     *
+     * @param {Object} options
+     * @param {File} options.file
+     * @param {string} options.folderId - Untyped folder id (e.g. no "folder_" prefix)
+     * @param {string} [options.fileId] - Untyped file id (e.g. no "file_" prefix)
+     * @param {string} options.sessionId
+     * @param {Function} [options.errorCallback]
+     * @param {Function} [options.progressCallback]
+     * @param {Function} [options.successCallback]
+     * @return {void}
+     */
+    resume({
+        file,
+        folderId,
+        errorCallback,
+        progressCallback,
+        sessionId,
+        successCallback,
+        overwrite = true,
+        fileId,
+    }: {
+        errorCallback?: Function,
+        file: File,
+        fileId: ?string,
+        folderId: string,
+        overwrite?: boolean,
+        progressCallback?: Function,
+        sessionId: string,
+        successCallback?: Function,
+    }): void {
+        this.setFileInfo({ file, folderId, errorCallback, progressCallback, successCallback, overwrite, fileId });
+        this.sessionId = sessionId;
+
+        if (!this.sha1Worker) {
+            this.sha1Worker = createWorker();
+        }
+        this.sha1Worker.addEventListener('message', this.onWorkerMessage);
+
+        this.getSessionInfo();
+    }
+
+    /**
+     * Get session information from API.
+     * Uses session info to commit a complete session or continue an in-progress session.
+     *
+     * @private
+     * @return {void}
+     */
+    getSessionInfo = async (): Promise<any> => {
+        const uploadUrl = this.getBaseUploadUrl();
+        const sessionUrl = `${uploadUrl}/files/upload_sessions/${this.sessionId}`;
+        try {
+            const response = await this.xhr.get({ url: sessionUrl });
+            this.getSessionSuccessHandler(response.data);
+        } catch (error) {
+            this.getSessionErrorHandler(error);
+        }
+    };
+
+    /**
+     * Handles a getSessionInfo success and either commits the session or continues to process
+     * the parts that still need to be uploaded.
+     *
+     * @param response
+     * @return {void}
+     */
+    getSessionSuccessHandler(data: any): void {
+        const { total_parts, part_size, session_endpoints } = data;
```

Comment:
it worth checking "session_expires_at". if the session is already expired, you just can not resume the download

Reply:
From what I saw in the documentation for the chunked upload API, I thought that any request made with a session ID for a session that has expired would return a error, and so the scenario where the session has expired is being handled in the getSessionErrorHandler. Is this not necessarily the case that an error would be returned?

Reply:
if you know that your session is expired why not to optimize and fail (or restart the upload) immediately? yet if you feel adding an extra check for session_expires_at will hurt the code readability i would skip it!

it is up to you to decide, yet if you decided to add session_expires_at check, make sure NOT to remove the "scenario where the session has expired is being handled in the getSessionErrorHandler"